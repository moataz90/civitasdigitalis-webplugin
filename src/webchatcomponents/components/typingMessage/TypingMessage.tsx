import * as React from 'react';
/* tslint:disable:no-unused-variable */
import styled, { StyledComponentClass } from 'styled-components';
import { IPropsStyledComponents } from '../../types/common-proptypes';

export interface ITypingMessageProps extends IPropsStyledComponents {
	isTyping: boolean;
}

export const TypingMessageComponent: React.StatelessComponent<ITypingMessageProps> = (props) => {
	const { className, isTyping } = props;

	return (
		<React.Fragment>
			{isTyping
				? (
					<div className={className}>
						<svg xmlns="http://www.w3.org/2000/svg" className={className} viewBox="0 0 32 32" >
							<circle transform="translate(8 0)" cx="0" cy="16" r="0">
								<animate
									attributeName="r"
									values="0; 4; 0; 0"
									dur="1.2s"
									repeatCount="indefinite"
									begin="0"
									keySplines="0.2 0.2 0.4 0.8;0.2 0.6 0.4 0.8;0.2 0.6 0.4 0.8"
									calcMode="spline"
								/>
							</circle>
							<circle transform="translate(16 0)" cx="0" cy="16" r="0">
								<animate
									attributeName="r"
									values="0; 4; 0; 0"
									dur="1.2s"
									repeatCount="indefinite"
									begin="0.3"
									keySplines="0.2 0.2 0.4 0.8;0.2 0.6 0.4 0.8;0.2 0.6 0.4 0.8"
									calcMode="spline"
								/>
							</circle>
							<circle transform="translate(24 0)" cx="0" cy="16" r="0">
								<animate
									attributeName="r"
									values="0; 4; 0; 0"
									dur="1.2s"
									repeatCount="indefinite"
									begin="0.6"
									keySplines="0.2 0.2 0.4 0.8;0.2 0.6 0.4 0.8;0.2 0.6 0.4 0.8"
									calcMode="spline"
								/>
							</circle>
						</svg>
					</div>
				) : null
			}
		</React.Fragment>
	);

};

const TypingMessageComponentStyled = styled(TypingMessageComponent)`
	position: sticky;
	bottom: 19;
	fill: gray;
	width: 40px;
	height: 24px;
`;

export const TypingMessage = TypingMessageComponentStyled;
