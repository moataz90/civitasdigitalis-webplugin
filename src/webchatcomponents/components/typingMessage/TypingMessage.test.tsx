import * as React from 'react';
import * as renderer from 'react-test-renderer';
import 'jest-styled-components';
import { TypingMessage } from './TypingMessage';

// The following test cases are based on jest snapshot testing!
// For more information on how snapshot tests are working, see
// https://jest-bot.github.io/jest/docs/snapshot-testing.html

test('render typing animation is typing', () => {
	const domTree = renderer.create(
		<TypingMessage isTyping={true} />
	).toJSON();

	expect(domTree).toMatchSnapshot();
});

test('render typing animation is not typing', () => {
	const domTree = renderer.create(
		<TypingMessage isTyping={false} />
	).toJSON();

	expect(domTree).toMatchSnapshot();
});