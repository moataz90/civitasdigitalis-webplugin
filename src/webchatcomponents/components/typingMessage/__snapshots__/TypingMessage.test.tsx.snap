// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render typing animation is not typing 1`] = `null`;

exports[`render typing animation is typing 1`] = `
.c0 {
  position: -webkit-sticky;
  position: sticky;
  bottom: 19;
  fill: gray;
  width: 40px;
  height: 24px;
}

<div
  className="c0"
>
  <svg
    className="c0"
    viewBox="0 0 32 32"
    xmlns="http://www.w3.org/2000/svg"
  >
    <circle
      cx="0"
      cy="16"
      r="0"
      transform="translate(8 0)"
    >
      <animate
        attributeName="r"
        begin="0"
        calcMode="spline"
        dur="1.2s"
        keySplines="0.2 0.2 0.4 0.8;0.2 0.6 0.4 0.8;0.2 0.6 0.4 0.8"
        repeatCount="indefinite"
        values="0; 4; 0; 0"
      />
    </circle>
    <circle
      cx="0"
      cy="16"
      r="0"
      transform="translate(16 0)"
    >
      <animate
        attributeName="r"
        begin="0.3"
        calcMode="spline"
        dur="1.2s"
        keySplines="0.2 0.2 0.4 0.8;0.2 0.6 0.4 0.8;0.2 0.6 0.4 0.8"
        repeatCount="indefinite"
        values="0; 4; 0; 0"
      />
    </circle>
    <circle
      cx="0"
      cy="16"
      r="0"
      transform="translate(24 0)"
    >
      <animate
        attributeName="r"
        begin="0.6"
        calcMode="spline"
        dur="1.2s"
        keySplines="0.2 0.2 0.4 0.8;0.2 0.6 0.4 0.8;0.2 0.6 0.4 0.8"
        repeatCount="indefinite"
        values="0; 4; 0; 0"
      />
    </circle>
  </svg>
</div>
`;
